nextcord/client.py

```diff
@@ -2023,33 +2039,171 @@ def get_application_command(self, command_id: int) -> Optional[BaseApplicationCo
+    def get_application_command_from_signature(
+        self, *args: Any, search_locales: bool = False, **kwargs: Any
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        REQUIRED_ARGS = ("qualified_name", "type", "guild_id")
+        FUNC = "get_application_command_from_signature()"
+        ARGS_DEPRECATED_MESSAGE = "The signature passed to {name} is deprecated, please use keyword arguments only instead. Converted signature: {converted_signature}."
+        KWARG_DEPRECATION_MESSAGE = 'The keyword argument "{old}=" used in {name} is deprecated, please use "{new}=" instead.'
+
+        kwargs_convert_table: Dict[str, Any] = {"name": "qualified_name", "cmd_type": "type"}
+        VALID_KWARGS = ("type", "qualified_name", "guild_id", "search_locales")
+        for kw in kwargs.copy():
+            if kw in kwargs_convert_table:
+                warnings.warn(
+                    KWARG_DEPRECATION_MESSAGE.format(
+                        name=FUNC, old=kw, new=kwargs_convert_table[kw]
+                    ),
+                    stacklevel=2,
+                    category=FutureWarning,
+                )
+                kwargs[kwargs_convert_table[kw]] = kwargs.pop(kw)
+            elif kw not in VALID_KWARGS:
+                raise TypeError(f"{FUNC} got an unexpected keyword argument '{kw}'")
+
+        if len(args) > 3:
+            raise TypeError(f"{FUNC} takes 3 positional argument but {len(args)} were given")
+        for index, arg in enumerate(args):
+            if index in (0, 1):
+                if isinstance(arg, str):
+                    kwargs["qualified_name"] = kwargs.get("qualified_name", arg)
+                elif isinstance(arg, (int, ApplicationCommandType)):
+                    kwargs["type"] = kwargs.get("type", arg)
+            elif index == 2:
+                kwargs["guild_id"] = kwargs.get("guild_id", arg)
+
+        # mimicking the python error
+        def ARGS_ERROR_MESSAGE(*required_arguments: str) -> str:
+            base = f"{FUNC} missing {len(required_arguments)} required positional {'arguments' if len(required_arguments) > 1 else 'argument'}: "
+            if len(required_arguments) == 1:
+                base += f"`{required_arguments[0]}`"
+            elif len(required_arguments) == 2:
+                base += f"`{required_arguments[0]}` and `{required_arguments[1]}`"
+            else:
+                without_last_argument = required_arguments[:-1]
+                last_argument = required_arguments[-1]
+                base += (
+                    ", ".join(f"'{key}'" for key in without_last_argument)
+                    + f", and '{last_argument}'"
+                )
+
+            return base
+
+        MISSING_ARGS = []
+        for arg in REQUIRED_ARGS:
+            if arg not in kwargs:
+                MISSING_ARGS.append(arg)
+
+        if MISSING_ARGS:
+            message = ARGS_ERROR_MESSAGE(*MISSING_ARGS)
+            raise TypeError(message)
```

Comment:
Couldn't you just raise with direct result of `ARGS_ERROR_MESSAGE` instead of putting it in a variable?